devices/raspizerow/I2s/main.cpp

```diff
@@ -0,0 +1,301 @@
+        {
+            std::cerr << "Error from read: " << snd_strerror(rc) << std::endl;
+            break; // Exit the loop on error
+        }
+        else if (rc != (int)period_size)
+        {
+            std::cerr << "Short read, read " << rc << " frames" << std::endl;
+        }
+        else
+        {
+            // Append the captured data to the accumulated buffer
+            accumulatedBuffer.insert(accumulatedBuffer.end(), buffer.begin(), buffer.begin() + rc * bytesPerSample * channels);
+
+            if (accumulatedBuffer.size() >= targetBytes)
+            {
+                audioQueue.push(accumulatedBuffer);
+                accumulatedBuffer.clear();
+            }
+        }
+    }
+}
+
+void handleAudioBuffer()
+{
+    while (running)
+    {
+        std::vector<char> dataChunk;
+        // Accumulate enough data to form a complete WAV file
+        while (dataChunk.size() < targetBytes)
+        {
+            std::vector<char> buffer = audioQueue.pop();
+            dataChunk.insert(dataChunk.end(), buffer.begin(), buffer.end());
+        }
+
+        // Process and send the accumulated data
+        if (!dataChunk.empty())
+        {
+            std::ofstream wavFile("output.wav", std::ios::binary);
+            int bitsPerSample = 32;
+            int dataSize = dataChunk.size();
+            writeWavHeader(wavFile, bitsPerSample, dataSize);
+            wavFile.write(dataChunk.data(), dataSize);
+            wavFile.close();
+
+            // Send the WAV file to the server
+            sendWav("output.wav");
+
+            // Delete the file after sending
+            std::filesystem::remove("output.wav");
+        }
+    }
+
+    // After exiting the loop, check if there's any remaining data to be processed
+    // This could be less than targetBytes but should still be sent
+    if (!audioQueue.empty())
+    {
+        std::vector<char> remainingData;
+        while (!audioQueue.empty())
+        {
+            std::vector<char> buffer = audioQueue.pop();
+            remainingData.insert(remainingData.end(), buffer.begin(), buffer.end());
+        }
+
+        if (!remainingData.empty())
+        {
+            std::ofstream wavFile("output_remaining.wav", std::ios::binary);
+            int bitsPerSample = 32;
+            int dataSize = remainingData.size();
+            writeWavHeader(wavFile, bitsPerSample, dataSize);
+            wavFile.write(remainingData.data(), dataSize);
+            wavFile.close();
+
+            // Send the remaining WAV file to the server
+            sendWav("output_remaining.wav");
+
+            // Delete the file after sending
+            std::filesystem::remove("output_remaining.wav");
+        }
+    }
+}
+
+int main()
+{
+    snd_pcm_t *capture_handle;
+    snd_pcm_format_t format = SND_PCM_FORMAT_S32_LE;
+
+    signal(SIGINT, signalHandler);
+
+    // Open PCM device for recording
+    rc = snd_pcm_open(&capture_handle, "plughw:0", SND_PCM_STREAM_CAPTURE, 0);
```

Comment:
Is this how you communicate with the GPIO?

Reply:
ignore that, got my answer from here https://pinout.xyz/pinout/pcm

<img width="688" alt="image" src="https://github.com/adamcohenhillel/ADeus/assets/633840/396b7bce-92e7-42b1-86c0-4d4bf894a07e">

Reply:
Does the pcm comes from the package `#include <alsa/asoundlib.h>`?

Reply:
Yes the "plughw:0" is how the device is accessed. The 0 in the id.

To make it better we can look into how this could be automatically detected and filled in.